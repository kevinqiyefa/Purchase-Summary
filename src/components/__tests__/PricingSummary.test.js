import React from 'react';
import { shallow } from 'enzyme';
import PricingSummary from '../PriceSummary';
import toJson from 'enzyme-to-json';

describe('Testing PricingSummary Component', () => {
  let wrapped;
  const pricing = {
    subtotal: 102.96,
    savings: 3.85,
    tax: 8.92,
    total: 108.03,
    zip: 85050
  };

  beforeEach(() => {
    wrapped = shallow(<PricingSummary pricing={pricing} />);
  });

  it('shows 5 children in the DIV', () => {
    expect(wrapped.find('.Price-Summary-Main').children().length).toEqual(5);
  });

  it('matches snapshot', () => {
    const serialized = toJson(wrapped);
    expect(serialized).toMatchSnapshot();
  });
});
