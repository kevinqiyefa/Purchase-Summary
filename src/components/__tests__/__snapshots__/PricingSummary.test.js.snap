// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PricingSummary Component matches snapshot 1`] = `
<div
  className="Price-Summary-Main"
>
  <div>
    <span>
      Subtotal
    </span>
    <span
      className="bold-text"
    >
      $
      102.96
    </span>
  </div>
  <div>
    <div
      className="Price-Summary-Savings"
    >
      Pickup savings
      <span
        className="tooltiptext"
      >
        Pickup your order in the store helps cut cost, and we pass the savings on to you.
      </span>
    </div>
    <span
      className="red-text bold-text"
    >
      -$
      3.85
    </span>
  </div>
  <div>
    <div
      className="Price-Summary-Tax-Zip"
    >
      <span>
        Est. taxes & fees
      </span>
      <span>
        (Based on 
        85050
        )
      </span>
    </div>
    <span
      className="bold-text"
    >
      $
      8.92
    </span>
  </div>
  <hr />
  <div
    className="big-text"
  >
    <span
      className="bold-text"
    >
      Est. total
    </span>
    <span
      className="bold-text"
    >
      $
      108.03
    </span>
  </div>
</div>
`;
